homeassistant/components/sensor/mold_indicator.py

```diff
@@ -41,66 +44,125 @@
-                 indoor_humidity_sensor, calib_factor):
+    def __init__(self, hass, name, indoor_temp_sensor,
+                 outdoor_temp_sensor, indoor_humidity_sensor, calib_factor):
         """Initialize the sensor."""
+        self.hass = hass
         self._state = None
         self._name = name
         self._indoor_temp_sensor = indoor_temp_sensor
         self._indoor_humidity_sensor = indoor_humidity_sensor
         self._outdoor_temp_sensor = outdoor_temp_sensor
         self._calib_factor = calib_factor
         self._is_metric = hass.config.units.is_metric
+        self._available = False
+        self._entities = set([self._indoor_temp_sensor,
+                              self._indoor_humidity_sensor,
+                              self._outdoor_temp_sensor])
 
         self._dewpoint = None
         self._indoor_temp = None
         self._outdoor_temp = None
         self._indoor_hum = None
         self._crit_temp = None
 
-        track_state_change(hass, indoor_temp_sensor, self._sensor_changed)
-        track_state_change(hass, outdoor_temp_sensor, self._sensor_changed)
-        track_state_change(hass, indoor_humidity_sensor, self._sensor_changed)
+    async def async_added_to_hass(self):
+        """Register callbacks."""
+        @callback
+        def mold_indicator_sensors_state_listener(entity, old_state,
+                                                  new_state):
+            """Handle for state changes for dependent sensors."""
+            _LOGGER.debug("Sensor state change for %s that had old state %s "
+                          "and new state %s", entity, old_state, new_state)
 
-        # Read initial state
-        indoor_temp = hass.states.get(indoor_temp_sensor)
-        outdoor_temp = hass.states.get(outdoor_temp_sensor)
-        indoor_hum = hass.states.get(indoor_humidity_sensor)
+            if self._update_sensor(entity, old_state, new_state):
+                self.async_schedule_update_ha_state(True)
 
-        if indoor_temp:
-            self._indoor_temp = MoldIndicator._update_temp_sensor(indoor_temp)
+        @callback
+        def mold_indicator_startup(event):
+            """Add listeners and get 1st state."""
+            _LOGGER.debug("Startup for %s", self.entity_id)
 
-        if outdoor_temp:
-            self._outdoor_temp = MoldIndicator._update_temp_sensor(
-                outdoor_temp)
+            async_track_state_change(self.hass, self._entities,
+                                     mold_indicator_sensors_state_listener)
 
-        if indoor_hum:
-            self._indoor_hum = MoldIndicator._update_hum_sensor(indoor_hum)
+            # Read initial state
+            indoor_temp = self.hass.states.get(self._indoor_temp_sensor)
+            outdoor_temp = self.hass.states.get(self._outdoor_temp_sensor)
+            indoor_hum = self.hass.states.get(self._indoor_humidity_sensor)
+
+            schedule_update = True if self._update_sensor(
```

Comment:
```py
schedule_update = self._update_sensor(...)
```

Reply:
Was trying to be too smart for my own good here. :-) Changed.